x/superfluid/keeper/twap_price.go

```diff
@@ -34,7 +37,21 @@ func (k Keeper) SetOsmoEquivalentMultiplier(ctx sdk.Context, epoch int64, denom
 	prefixStore.Set([]byte(denom), bz)
 }
 
-func (k Keeper) GetSuperfluidOSMOTokens(ctx sdk.Context, denom string, amount osmomath.Int) (osmomath.Int, error) {
+func isNative(denom string) bool {
+	if strings.HasPrefix(denom, cltypes.ConcentratedLiquidityTokenPrefix) {
+		return false
+	}
+	if strings.HasPrefix(denom, gammtypes.GAMMTokenPrefix) {
+		return false
+	}
+	return true
+}
+
+func (k Keeper) GetSuperfluidOSMOTokens(ctx sdk.Context, denom string, amount osmomath.Int, includeNative bool) (osmomath.Int, error) {
+	if !includeNative && isNative(denom) {
+		return osmomath.ZeroInt(), nil
+	}
```

Comment:
Not sure I get what you mean here. Which part should be split? GetSuperfluidOSMOTokens?

If so, I think it's better to not split it. Otherwise we will have to have the includeNative logic everywhere where this is used. I'm open to be convinced otherwise, though

Reply:
Im saying instead of adding the boolean flag to GetSuperfluidOSMOTokens, there should be two methods, GetSuperfluidOSMOTokensWithoutNative and GetSuperfluidOSMOTokensWithNative.

Reply:
refactored in f9f09a14d94424bdc47d657b41f5dfefcc442d7c